portsyncd/portsyncd.cpp

```diff
@@ -103,16 +105,26 @@ int main(int argc, char **argv)
         {
             Selectable *temps;
             int ret;
-            ret = s.select(&temps, 1);
+            ret = s.select(&temps, DEFAULT_SELECT_TIMEOUT);
 
             if (ret == Select::ERROR)
             {
                 cerr << "Error had been returned in select" << endl;
                 continue;
             }
+            else if (ret == Select::TIMEOUT)
+            {
+                continue;
+            }
+            else if (ret != Select::OBJECT)
+            {
+                SWSS_LOG_ERROR("Unknown return value from Select %d", ret);
+                continue;
+            }
 
-            if (ret == Select::TIMEOUT)
+            if (temps == static_cast<Selectable*>(&netlink))
```

Comment:
>if (temps == static_cast<Selectable*>(&netlink)) [](start = 12, length = 48)

The behavior is changed and suspicious? Could you focus on "fix default select timeout" in this PR?

Reply:
What is suspicious here? On netlink message, check wether we need to send PortInitDone. Previously this check was done every 1ms.

Reply:
It not necessarily mean your change is wrong. I mean the purpose of this PR is to "fix default select timeout".

Let's assume if you don't change the behavior here, this check will be done every 1000ms. Anything wrong? If there is a bug, I prefer fix it in a separate PR.

---
In reply to: [420307407](https://github.com/Azure/sonic-swss/pull/1287#discussion_r420307407) [](ancestors = 420307407)

Reply:
Yes, it is kind of another problem. That's why I assume the select timeout was set to very small value. If we increase select timeout to 1 sec, the check will be performed every 1 sec. This means there might be 1 sec delay in notifying system about PortInitDone. To avoid this unneccesary delay, we check if select was unblocked due to netlink message and do the PortInitDone check. So, I think that issue is strongly related to select timeout and don't see a reason to fix it in a seperate PR. I hope that sounds reasonable to fix in one PR

Reply:
`g_portSet` is decreased by LinkSync::onMsg(). Not sure the relation ship between the timing of that and selecting `netlink`. Is it possible to have time risk?

---
In reply to: [421040960](https://github.com/Azure/sonic-swss/pull/1287#discussion_r421040960) [](ancestors = 421040960,420307407)

Reply:
LinkSync::onMsg is called in NetLink::readData() in select(), so g_portSet will be updated after select()